dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationResultImpl.java

```diff
@@ -6,17 +6,27 @@
  */
 package com.powsybl.dynamicsimulation;
 
+import java.util.Collections;
+import java.util.Map;
+
+import com.powsybl.timeseries.StringTimeSeries;
+import com.powsybl.timeseries.TimeSeries;
+
 /**
  * @author Marcos de Miguel <demiguelm at aia.es>
  */
 public class DynamicSimulationResultImpl implements DynamicSimulationResult {
 
     private final boolean status;
     private final String logs;
+    private final Map<String, TimeSeries> curves;
+    private final StringTimeSeries timeLine;
 
-    public DynamicSimulationResultImpl(boolean status, String logs) {
+    public DynamicSimulationResultImpl(boolean status, String logs, Map<String, TimeSeries> curves, StringTimeSeries timeLine) {
         this.status = status;
         this.logs = logs;
+        this.curves = curves;
+        this.timeLine = timeLine;
```

Comment:
Could you assert that both curves and timeline are not null? I saw in the unit test that the timeline could be null. Is it possible to create an empty timeline without any entries?

Reply:
Done.